src/main/java/org/jabref/gui/externalfiles/AutoSetFileLinksUtil.java

```diff
@@ -127,16 +139,60 @@ public List<LinkedFile> findAssociatedNotLinkedFiles(BibEntry entry) throws IOEx
 
             if (!fileAlreadyLinked) {
                 Optional<ExternalFileType> type = FileUtil.getFileExtension(foundFile)
-                                                            .map(extension -> ExternalFileTypes.getExternalFileTypeByExt(extension, filePreferences))
-                                                            .orElse(Optional.of(new UnknownExternalFileType("")));
+                                                          .map(extension -> ExternalFileTypes.getExternalFileTypeByExt(extension, filePreferences))
+                                                          .orElse(Optional.of(new UnknownExternalFileType("")));
 
                 String strType = type.isPresent() ? type.get().getName() : "";
                 Path relativeFilePath = FileUtil.relativize(foundFile, directories);
                 LinkedFile linkedFile = new LinkedFile("", relativeFilePath, strType);
                 linkedFiles.add(linkedFile);
             }
         }
-
         return linkedFiles;
     }
+
+    public RelinkedResults relinkingFiles(List<LinkedFile> listlinked) throws FileNotFoundException {
+        List<LinkedFile> changedFiles = new ArrayList<>();
+        List<String> exceptions = new ArrayList<>();
+
+        for (LinkedFile file : listlinked) {
+            Path path = Path.of(file.getLink());
+            if (!Files.exists(path)) {
+                Path filePath = Path.of(file.getLink());
+                // May need to put some limit since it can cause considerable performance problems.
+                String directoryPath = filePath.getParent().getParent().toString();
+
+                String fileNameString = filePath.getFileName().toString();
+
+                List<Path> fileLocations = searchFileInDirectoryAndSubdirectories(Path.of(directoryPath), fileNameString);
+                if (!fileLocations.isEmpty()) {
+                    // File locations is a list but as stated in the link below, it is a rare case. But can be solved if time allows.
+                    // https://github.com/JabRef/jabref/issues/9798#issuecomment-1524155132
+                    file.setLink(fileLocations.get(0).toString());
+                    changedFiles.add(file);
+                } else {
+                    exceptions.add(fileNameString);
+                    throw new FileNotFoundException();
```

Comment:
No exception here. Just leave the link as is.

Exceptions are huge derivations from the normal results.

See, if I have a library wtih 10000 PDFs, moved 20 files and 19 can be relinked, the whole processing should not be stopped if one file cannot be found.

It is OK to report a list of files which could not be relinked. If this is hard, then just log the non-found files.